Type editor: indicate link entity types, bump version

CiaranMn opened the pull request:
## 🌟 What is the purpose of this PR?

- Bump the version used of the type editor to bring in improvements made in https://github.com/hashintel/hash/pull/2123
- Bug fixes in updating option state
- Indicate link entity types

<img width="706" alt="Screenshot 2023-02-24 at 23 13 16" src="https://user-images.githubusercontent.com/37743469/221320740-1f2f11b7-f3b3-4a9a-b751-6284525977b4.png">

luisbettencourt commented on apps/site/src/pages/[shortname]/types/entity-type/[...slug-maybe-version].page.tsx:
```diff
@@ -187,6 +195,19 @@ const EntityTypePage: NextPage = () => {
               </Link>
               <Stack flexDirection="row" alignItems="center">
+                {isLinkEntityType && (
+                  <Tooltip
```
This tooltip isn't working. To fix it you can wrap it in a `Box`

CiaranMn commented:
> This tooltip isn't working. To fix it you can wrap the icon inside a Box component.

@luisbettencourt thank you, now fixed. I've also added the `react-hook-form` patch we have in `hash`.

I'm going to merge this as I need the bug fixes in to work on tests
